web/src/pages/projectInsights/Details.tsx

```diff
@@ -0,0 +1,239 @@
+// Details.tsx
+import React, { useState, useEffect } from 'react'
+import { ProjectInsightsDetails, ProjectApi, ProjectInsightsApi, EnumsApi } from '../../sm-api'
+import { Button } from 'semantic-ui-react'
+import ProjectAndSeqTypeSelector from './ProjectAndSeqTypeSelector'
+import DetailsTable from './DetailsTable'
+
+interface SelectedProject {
+    id: number
+    name: string
+}
+
+const Details: React.FC = () => {
+    // Get the list of  projects from the project API
+    const [allData, setAllData] = useState<ProjectInsightsDetails[]>([])
+    const [projectNames, setProjectNames] = React.useState<string[]>([])
```

Comment:
@dancoates, can you take a look at the typescript + react on this function, to see if you have suggestions on improving the code here